libraries/libraries_api.go

```diff
@@ -177,85 +25,42 @@ func NewLibraryFromInstanceState(i any) (lib Library) {
 		lib.Maven.Coordinates, _ = maven["coordinates"].(string)
 		lib.Maven.Repo, _ = maven["repo"].(string)
 	}
 	cranList, ok := raw["cran"].([]any)
 	if ok && len(cranList) == 1 {
-		lib.Cran = &Cran{}
+		lib.Cran = &compute.RCranLibrary{}
 		cran := cranList[0].(map[string]any)
 		lib.Cran.Package, _ = cran["package"].(string)
 		lib.Cran.Repo, _ = cran["repo"].(string)
 	}
 	return lib
 }
 
-// Library is a construct that contains information of the location of the library and how to download it
-type Library struct { // TODO: discuss if we can make a dedicated entity just for terraform...
-	Jar   string `json:"jar,omitempty" tf:"group:lib"`
-	Egg   string `json:"egg,omitempty" tf:"group:lib"`
-	Whl   string `json:"whl,omitempty" tf:"group:lib"`
-	Pypi  *PyPi  `json:"pypi,omitempty" tf:"group:lib"`
-	Maven *Maven `json:"maven,omitempty" tf:"group:lib"`
-	Cran  *Cran  `json:"cran,omitempty" tf:"group:lib"`
-}
-
-func (library Library) String() string {
-	if library.Whl != "" {
-		return fmt.Sprintf("whl:%s", library.Whl)
-	}
-	if library.Jar != "" {
-		return fmt.Sprintf("jar:%s", library.Jar)
-	}
-	if library.Pypi != nil && library.Pypi.Package != "" {
-		return fmt.Sprintf("pypi:%s%s", library.Pypi.Repo, library.Pypi.Package)
-	}
-	if library.Maven != nil && library.Maven.Coordinates != "" {
-		mvn := library.Maven
-		return fmt.Sprintf("mvn:%s%s%s", mvn.Repo, mvn.Coordinates,
-			strings.Join(mvn.Exclusions, ""))
-	}
-	if library.Egg != "" {
-		return fmt.Sprintf("egg:%s", library.Egg)
-	}
-	if library.Cran != nil && library.Cran.Package != "" {
-		return fmt.Sprintf("cran:%s%s", library.Cran.Repo, library.Cran.Package)
-	}
-	return "unknown"
-}
-
-func (library Library) GetID(clusterId string) string {
-	return fmt.Sprintf("%s/%s", clusterId, library.String())
-}
-
-// ClusterLibraryList is request body for install and uninstall
-type ClusterLibraryList struct {
-	ClusterID string    `json:"cluster_id,omitempty" url:"cluster_id,omitempty"`
-	Libraries []Library `json:"libraries,omitempty" tf:"slice_set,alias:library"`
-}
-
 // Diff returns install/uninstall lists given a cluster lib status
-func (cll *ClusterLibraryList) Diff(cls ClusterLibraryStatuses) (ClusterLibraryList, ClusterLibraryList) {
-	inConfig := map[string]Library{}
+func GetLibrariesToInstallAndUninstall(cll LibraryList, cls *compute.ClusterLibraryStatuses) (compute.InstallLibraries, compute.InstallLibraries) {
+	inConfig := map[string]compute.Library{}
 	for _, lib := range cll.Libraries {
 		inConfig[lib.String()] = lib
 	}
-	inState := map[string]Library{}
+	inState := map[string]compute.Library{}
 	for _, status := range cls.LibraryStatuses {
 		lib := *status.Library
 		inState[lib.String()] = lib
 	}
-	toInstall := ClusterLibraryList{ClusterID: cll.ClusterID}
-	toUninstall := ClusterLibraryList{ClusterID: cll.ClusterID}
+	toInstall := compute.InstallLibraries{ClusterId: cll.ClusterId}
+	// TODO: UninstallLibraries doesn't have sort() method
```

Comment:
You don't need a sort method to be defined on the type. You can use `slices.SortFunc()`. But why not just return `([]Library, []Library)`?